docs/components/table-view.md

```diff
@@ -0,0 +1,1039 @@
+    label: 'Row actions',
+  },
+]
+
+const tableData = ref<TableViewData>([
+  {
+    id: 1,
+    // notice that property keys in data object correspond to each respective key in headers const
+    name: 'Leanne Graham', 
+    username: 'Bret',
+    email: 'Sincere@april.biz'
+  },
+  {
+    id: 2,
+    name: 'Ervin Howell',
+    username: 'Antonette',
+    email: 'Shanna@melissa.tv'
+  },
+  ...
+])
+
+const sortData = (sortData: TableSortPayload): void => {
+  const data = [...tableData.value]
+  const { sortColumnKey, sortColumnOrder } = sortData || { sortColumnKey: 'username', sortColumnOrder: 'asc' }
+
+  data.sort((a: Record<string, any>, b: Record<string, any>) => {
+    if (sortColumnKey === 'username') {
+      if (sortColumnOrder === 'asc') {
+        if (a.username > b.username) {
+          return 1
+        } else if (a.username < b.username) {
+          return -1
+        }
+
+        return 0
+      } else {
+        if (a.username > b.username) {
+          return -1
+        } else if (a.username < b.username) {
+          return 1
+        }
+
+        return 0
+      }
+    }
+
+    return 0
+  })
+
+  tableData.value = data
+}
+</script>
+```
+
+:::tip NOTE
+Notice that in the example above the _Username_ column is `sortable` and the _Email_ column is `hidable`.
+:::
+
+### loading
+
+Boolean to control whether the component should display the loading state. Defaults to `false`.
+
+<KTableView
+  loading
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  loading
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### error
+
+Boolean to control whether the component should display the error state. Defaults to `false`. See [error state](#error-1) section for more customization options.
+
+<KTableView
+  error
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  error
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### resizeColumns
+
+Allow table column width to be resizable (defaults to `false`). Adjusting a column's width will trigger an [`update:table-preferences` event](#updatetable-preferences).
+
+<KTableView
+  resize-columns
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  resize-columns
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### rowHover
+
+Boolean to control whether table should display hover state upon hovering rows. Defaults to `true`.
+
+### tablePreferences
+
+Table preferences object for some configuration options.
+
+```ts
+interface TablePreferences {
+  pageSize?: number // the number of items to display per page
+  sortColumnKey?: string // the column to sort by's `key` defined in the table headers
+  sortColumnOrder?: SortColumnOrder // the order by which to sort the column, one of `asc` or `desc`
+  columnWidths?: Record<string, number> // the customized column widths, if resizing is allowed
+  columnVisibility?: Record<string, boolean> // column visibility, if visibility is toggleable
```

Comment:
nit: can you move the comments above each property so you can read them without having to scroll horizontally?

Reply:
Done